core/src/main/scala/org/http4s/internal/Random.scala

```diff
@@ -67,21 +68,43 @@ private[http4s] object Random {
     * non-blocking.  If a non-blocking instance can't be guaranteed,
     * falls back to a blocking implementation.
     */
-  def javaSecuritySecureRandom[F[_]](blocker: Blocker)(implicit F: Sync[F], cs: ContextShift[F]) =
+  def javaSecuritySecureRandom[F[_]](blocker: Blocker)(implicit F: Sync[F], cs: ContextShift[F]) = {
+    // This is a known, non-blocking, threadsafe algorithm
+    def happyRandom = F.delay(SecureRandom.getInstance("NativePRNGNonBlocking"))
```

Comment:
So in what situations will this algorithm not be available? Are any of the other implementations applicable? E.g. `NativePRNG` seems non-blocking after it has been seeded.
```
SHA1PRNG (Initial seeding is currently done via a combination of system attributes and the java.security entropy gathering device)
NativePRNG (nextBytes() uses /dev/urandom, generateSeed() uses /dev/random)
NativePRNGBlocking (nextBytes() and generateSeed() use /dev/random)
NativePRNGNonBlocking (nextBytes() and generateSeed() use /dev/urandom)
```
https://docs.oracle.com/javase/8/docs/technotes/guides/security/SunProviders.html#SUNProvider

Reply:
I don't think it will be available on Windows.  The NativePRNG on Windows

Reply:
I don't think `NativePRNGNonBlocking` will be available on Windows, whose `NativePRNG` is a [dummy implementation](https://github.com/openjdk/jdk/blob/master/src/java.base/windows/classes/sun/security/provider/NativePRNG.java).  It also won't be available if `/dev/urandom` is unreadable, which would be highly weird.

`NativePRNG` should be available in mostly the same circumstances as `NativePRNGNonBlocking`.

I can't find any guarantees about `SHA1PRNG`.  Sounds like the best case is that we'd have to block on seeding.